viewer/event_db.cpp

```diff
@@ -44,7 +55,24 @@ std::vector<EventRef> EventDB::get_data(HT_TimestampNs start_ts,
     }    
     else
     {
-        _query_event_klass(response, _events[query.klass_id], start_ts, stop_ts); 
+        if (_events.find(query.klass_id) == _events.end())
+            return response;
+
+        std::vector<parser::Event>::iterator first_event;
+        std::vector<parser::Event>::iterator last_event;
+        _get_range_of_events(_events[query.klass_id], start_ts, stop_ts, first_event, last_event);
+        --last_event;
```

Comment:
I'm not sure if it's safe; what if `last_event` equals `_events[query.klass_id].begin()`? It's undefined behavior, and you do an extra check here.